src/conditional_style.rs

```diff
@@ -0,0 +1,258 @@
+    fn from(value: &'a str) -> Self {
+        StarshipConditionalStyleConfig(StarshipConditionalStyle {
+            env: None,
+            operator: None,
+            expected_value: None,
+            style: value,
+        })
+    }
+}
+
+pub fn get_conditional_style<'a>(
+    context: &Context,
+    items: &[StarshipConditionalStyleConfig<'a>],
+) -> &'a str {
+    let mut conditional_style_values = items.iter().map(|x| &x.0);
+    let matching_style = conditional_style_values.find(|s| s.should_apply(context));
+    let last = conditional_style_values.last();
+
+    matching_style.or(last).map(|x| x.style).unwrap_or("")
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::context::{Context, Shell, Target};
+    use crate::serde_utils::ValueDeserializer;
+    use std::path::PathBuf;
+
+    fn create_context<'a>() -> Context<'a> {
+        Context::new_with_shell_and_path(
+            Default::default(),
+            Shell::Unknown,
+            Target::Main,
+            PathBuf::new(),
+            PathBuf::new(),
+        )
+    }
+
+    #[test]
+    fn test_exists_operator() {
+        let mut context = create_context();
+        let style = StarshipConditionalStyle {
+            env: Some("test"),
+            operator: Some(StarshipConditionalStyleOperator::Exists),
+            expected_value: None,
+            style: "",
+        };
+        assert!(!style.should_apply(&context));
+        context.env.insert("test", String::default());
+        assert!(style.should_apply(&context));
+    }
+
+    #[test]
+    fn test_equal_operator() {
+        let mut context = create_context();
+        let style = StarshipConditionalStyle {
+            env: Some("test"),
+            operator: Some(StarshipConditionalStyleOperator::Equal),
+            expected_value: Some("expected"),
+            style: "",
+        };
+
+        assert!(!style.should_apply(&context));
+        context.env.insert("test", String::from("not_expected"));
+        assert!(!style.should_apply(&context));
+        context.env.insert("test", String::from("expected"));
+        assert!(style.should_apply(&context));
+    }
+
+    #[test]
+    fn should_deserialize_from_string_value() {
+        let config = toml::value::Value::String(String::from("bold red dimmed"));
+        let deserializer = ValueDeserializer::new(&config);
+
+        let result = StarshipConditionalStyleConfig::deserialize(deserializer);
+
+        assert_eq!(
+            result,
+            Ok(StarshipConditionalStyleConfig(StarshipConditionalStyle {
+                env: None,
+                operator: None,
+                expected_value: None,
+                style: "bold red dimmed"
+            }))
+        );
+    }
+
+    #[test]
+    fn should_deserialize_from_table_value() {
+        let config = toml::toml! {
+            env = "HOSTNAME"
+            operator = "equal"
+            expected_value = "home"
```

Comment:
What if rather than all this, it just sticks to the API convention of using `when = ''' <script> '''` rather than these three values?

I think it would be a much better approach and keep the mechanics of other ways things are displayed on a condition. 

I expect this would mean that users would need to write their own small scripts that exit cleanly or naw, but that would be okay as those small scripts would be more useful outside of Starship.